src/ActionServer_FJT.c

```diff
@@ -503,7 +503,7 @@ void Ros_ActionServer_FJT_Goal_Complete(GOAL_END_TYPE goal_end_type)
             diff = fabs(feedback_FollowJointTrajectory.feedback.desired.positions.data[axis] - feedback_FollowJointTrajectory.feedback.actual.positions.data[axis]);
 
             double posTolerance = g_actionServer_FJT_SendGoal_Request.goal.goal_tolerance.data[axis].position; //user-provided a goal_tolerance
```

Comment:
should we not check whether `g_actionServer_FJT_SendGoal_Request.goal.goal_tolerance.data` is of the expected/required size? Otherwise we could be trying to read outside the array.

Reply:
> We might already be doing that before we get here, I haven't checked

We're not.

You're correct in that this is another potential issue.

@yai-rosejo mentioned in passing that he's got a branch doing some more thorough checking. I'll let him comment on what he's done.